### PR TITLE
[AIRFLOW-6685] ThresholdCheckOperator

### PR DESCRIPTION
**Pull request by alexzlue**
This PR includes a new operator in `CheckOperator` that allows users to perform a threshold data quality check.

`ThresholdCheckOperator` will check a single value, sql result against a threshold range, and will fail a task if it is outside this range. The lower and upper bound of the threshold can be defined as either a numeric values, or sql-statements that returns a numeric value.

---
Issue link: [AIRFLOW-6685](https://issues.apache.org/jira/browse/AIRFLOW-6685)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/operators/data_quality_check.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        raise AirflowException(body)
+
+    def _get_hook(self, conn_id):
+        """
+        _get_hook is a helper function for get_sql_value. Returns a database
+        hook depending on the conn_type and conn_id specified. Method will raise
+        an exception if hook is not supported.
+        """
+
+        conn_type = BaseHook.get_connection(conn_id).conn_type
```
Comment (mik-laj) [inline airflow/operators/data_quality_check.py]: Why didn't you use the Connection.get_hook method?

**Comment by eladkal** [comment]
In general the operators in this PR sounds like enhancement of [CheckOperator](https://github.com/apache/airflow/blob/master/airflow/operators/check_operator.py) 

1.  Why it needs to be in a new file?
2. Aren't CheckOperator and BaseDataQualityOperator have same functionality?

**Comment by alexzlue** [comment]
@eladkal Thanks for bringing this to mind. I do see that there is some functionality that I have that `CheckOperator` does not have. I will work on merging some of my work into this file then.

File: airflow/operators/check_operator.py [hunk 2]
```diff
@@ -330,3 +330,87 @@ def execute(self, context=None):
...
+                         f'Result: {round(meta_data.get("result"), 2)} is not within thresholds '
+                         f'{meta_data.get("min_threshold")} and {meta_data.get("max_threshold")}'
+                         )
+            raise AirflowException(error_msg)
+
+        self.log.info("Test %s Successful.", self.task_id)
+
+    def push(self, meta_data):
+        """
+        Optional: Send data check info and metadata to an external database.
```
Comment (eladkal) [inline airflow/operators/check_operator.py:408]: How can this be set?

**Review by eladkal — COMMENTED** [review]
Looks good.

File: airflow/operators/check_operator.py (hunk 2 above)
Comment (alexzlue) [reply airflow/operators/check_operator.py:408]: When inheriting from this class, push can be overwritten

**Review by eladkal — APPROVED** [review]

**Review by turbaszek — APPROVED** [review]

**Pull request merged**